### ISSUE TITLE
what is the rotation and translation of pytorch3d.ops.efficient_pnp output with respect to the inputs?

### ISSUE THREAD

**Issue by yhu9** [facebookresearch/pytorch3d#1057]
## ❓ Questions on how to use PyTorch3D

Hello, I'm trying to use the function: pytorch3d.ops.efficient_pnp

However, I don't understand what the Rotation and Translation matrix computed here represents.

The docs say this:

Efficient PnP algorithm [1] for Perspective-n-Points problem:
    finds a camera position (defined by rotation `R` and translation `T`) that
    minimizes re-projection error between the given 3D points `x` and
    the corresponding uncalibrated 2D points `y`, i.e. solves

    `y[i] = Proj(x[i] R[i] + T[i])`

However, the opencv implementation uses the EPnP algorithm for calibrated 2D points with known camera intrinsic. The paper for the EPnP algorithm is also for calibrated cameras. From what I can tell, the pytorch3d.ops.efficient_pnp function doesn't return the projection matrix used to compute the rotation and translation.

I would like to use the computed rotation and translation to project the 3D points x onto y, given the camera intrinsic I have for my FoVPerspectiveCameras. Any help to understand what the output Rotation and Translation of pytorch3d.ops.efficient_pnp would be appreciated. Any help on how I would project x to y given the R and T from the function would be even more appreciated.

**Comment by shapovalov** [comment]
Hello @yhu9 . The documentation further explains what we mean by `Proj`: it is just division by `z`. If your y’s are pixel coordinates, you have to “uncalibrate” them first to obtain points in camera coordinates, i.e. multiply by the inverse of the intrinsic calibration matrix. Hope that helps but feel free to reopen the issue if you need further clarifications.

**Comment by yhu9** [comment]
Although I find that the predicted rotation and translation is slightly off, with your comment I think I got it to work. For clarification, if y's are in pixel coordinates you're saying to define my intrinsic matrix and take its inverse and multiply with y in homogeneous coordinates like so.

`
# y is of size (b,n,2)
# x is of size (b,n,3)

K = torch.tensor([[f,0,px].[0,f,py],[0,0,1]]).repeat(b,1,1)
K_inv = torch.inverse(K).permute(0,2,1)

y_homo = torch.cat([y,torch.ones(b,n,1)],dim=-1)
y_uncalibrated = torch.bmm(y_homo,K_inv)
y_uncalibrated = y_uncalibrated / y_uncalibrated[:,:,:2].unsqueeze(2)

transform = pytorch3d.ops.efficient_pnp(x,y_uncalibrated)

# x[i] R[i] + T[i] = transform[0]
# R = transform[1]
# T = transform[2]

z = transform[0]
z = torch.bmm(z,K.permute(0,2,1))
z = z / z[:,:,2].unsqueeze(2)

`

I can confirm z is then roughly equal to y in image coordinates.
